### PR TITLE
fix: correctly implement REST patch operations

### LINKED ISSUE #78: Parsing metadata does not reach nested values
Unreachable nested value when parsing metadata in [`bucket.patch()`](https://github.com/googleapis/storage-testbench/blob/f51fbfe9ec5df612168838f844aa576d9b1744ba/gcs/bucket.py#L524), causing an internal server error 500

### PR DESCRIPTION
**Pull request by coryan**
Fixes #78

### CODE REVIEW

File: gcs/bucket.py [hunk 1]
```diff
@@ -480,6 +486,8 @@ def __update_metadata(self, source, update_mask):
         self.metadata.update_time.FromDatetime(datetime.datetime.now())
 
     def update(self, request, context):
+        # Bucket: patch only implemented for gRPC
```
[the remark this answers is not in the document]
Comment (coryan) [reply gcs/bucket.py:489]: Fixed, thanks.

File: testbench/common.py [hunk 2]
```diff
@@ -567,3 +567,38 @@ def preprocess_object_metadata(metadata):
...
+        if key not in patched:
+            if isinstance(subpatch, dict):
+                patched[key] = rest_patch({}, subpatch, location)
+            else:
+                patched[key] = subpatch
+            continue
+        subtarget = patched.get(key)
+        if isinstance(subtarget, dict) and isinstance(subpatch, dict):
+            patched[key] = rest_patch(subtarget, subpatch, location)
+        elif isinstance(subtarget, dict) == isinstance(subpatch, dict):
```
[the remark this answers is not in the document]
Comment (coryan) [reply testbench/common.py:600]: Not sure what is the question.  I think the code is correct and moderately easy to follow:

The first branch handles `isinstance(subtarget, dict) ==true and isinstance(subpatch, dict) == true`.
The next branch handles `isintance(subtarget, dict) == false and isinstance(subpatch, dict) == false` (if both are equal, but not both true, then both are false), but says that succinctly.
The last branch handles the cases where at least one, but not both of `isinstance(subpatch, dict) == true` or `isinstance(subtarget, dict) == true`, which is a conflict.

Does that make sense?  I added a comment for "future me", so we can understand it again.

[the remark this answers is not in the document]
Comment (cojenco) [reply testbench/common.py:600]: Yes that makes sense! I secretly added print statements to help my brain understand. Thanks again for the change!

**Review by cojenco — APPROVED** [review]
LGTM, thanks!

**Pull request merged**